addons/mail/static/src/discuss/core/common/attachment_model_patch.js

```diff
@@ -5,8 +5,12 @@ import { patch } from "@web/core/utils/patch";
 /** @type {import("models").Attachment} */
 const attachmentPatch = {
     get isDeletable() {
-        if (this.message && this.thread?.model === "discuss.channel") {
-            return this.message.editable;
+        if (this.thread?.model === "discuss.channel") {
+            return (
+                this.message_ids.some((m) => m.isSelfAuthored) ||
+                ["admin", "owner"].includes(this.thread?.self_member_id?.channel_role) ||
+                this.store.self_user?.is_admin
+            );
```

Comment:
Bad override pattern: this should return explicit True or False (depending on if we want to add a restriction or a permission) but in other cases it should still go towards the super call.

In particular it should be always True for non-internal users that have the ownership token.